File: src/main/java/ru/hollowhorizon/hc/common/objects/entities/TestEntity.java
package ru.hollowhorizon.hc.common.objects.entities;

import net.minecraft.entity.EntityType;
import net.minecraft.entity.ai.goal.LookAtGoal;
import net.minecraft.entity.ai.goal.LookRandomlyGoal;
import net.minecraft.entity.ai.goal.RandomWalkingGoal;
import net.minecraft.entity.monster.MonsterEntity;
import net.minecraft.entity.player.PlayerEntity;
import net.minecraft.potion.Effects;
import net.minecraft.world.World;
import ru.hollowhorizon.hc.api.utils.IAnimated;
import ru.hollowhorizon.hc.common.registry.ModEntities;
import ru.hollowhorizon.hc.client.render.entities.HollowAnimationManager;

public class TestEntity extends MonsterEntity implements IAnimated {
    private boolean isCustom;

    public TestEntity(World level) {
        super(ModEntities.testEntity, level);
    }

    public TestEntity(EntityType<? extends MonsterEntity> p_i48553_1_, World p_i48553_2_) {
        super(p_i48553_1_, p_i48553_2_);
    }

    @Override
    protected void registerGoals() {
        this.goalSelector.addGoal(8, new LookAtGoal(this, PlayerEntity.class, 8.0F));
        this.goalSelector.addGoal(8, new LookRandomlyGoal(this));
        this.goalSelector.addGoal(0, new RandomWalkingGoal(this, 0.3D));
    }

    @Override
    public void onAnimationUpdate(HollowAnimationManager manager) {
        if(this.hasEffect(Effects.GLOWING)) {
            manager.addAnimation("save_obelisk_activation", false);
        } else if (this.hasEffect(Effects.BLINDNESS)) {
            manager.addAnimation("save_obelisk_tick", false);
        }
    }
}

File: src/main/java/ru/hollowhorizon/hc/common/objects/entities/TestEntity.kt
package ru.hollowhorizon.hc.common.objects.entities

import net.minecraft.world.InteractionHand
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.EquipmentSlot
import net.minecraft.world.entity.PathfinderMob
import net.minecraft.world.entity.ai.goal.RandomLookAroundGoal
import net.minecraft.world.entity.ai.goal.RandomStrollGoal
import net.minecraft.world.item.ItemStack
import net.minecraft.world.item.Items
import net.minecraft.world.level.Level
import net.minecraft.world.level.pathfinder.BlockPathTypes
import ru.hollowhorizon.hc.client.models.gltf.Transform
import ru.hollowhorizon.hc.client.models.gltf.animations.AnimationType
import ru.hollowhorizon.hc.client.models.gltf.manager.AnimatedEntityCapability
import ru.hollowhorizon.hc.client.models.gltf.manager.IAnimated
import ru.hollowhorizon.hc.client.utils.get


class TestEntity(type: EntityType<TestEntity>, world: Level) : PathfinderMob(type, world), IAnimated {

    init {
        this[AnimatedEntityCapability::class].apply {
            model = "hc:models/entity/player_model.gltf"
            animations[AnimationType.IDLE] = "hello"
            transform = Transform.create {}
        }
        setPathfindingMalus(BlockPathTypes.WATER, -1.0f)
        this.setItemInHand(InteractionHand.MAIN_HAND, ItemStack(Items.NETHERITE_HOE))
        this.setItemInHand(InteractionHand.OFF_HAND, ItemStack(Items.TNT))

        this.setItemSlot(EquipmentSlot.FEET, ItemStack(Items.IRON_BOOTS))
        this.setItemSlot(EquipmentSlot.LEGS, ItemStack(Items.IRON_LEGGINGS))
        this.setItemSlot(EquipmentSlot.CHEST, ItemStack(Items.IRON_CHESTPLATE))
        this.setItemSlot(EquipmentSlot.HEAD, ItemStack(Items.IRON_HELMET))
    }

    override fun registerGoals() {
        super.registerGoals()
        this.goalSelector.addGoal(0, RandomLookAroundGoal(this))
        this.goalSelector.addGoal(1, RandomStrollGoal(this, 1.0, 10))
    }
}